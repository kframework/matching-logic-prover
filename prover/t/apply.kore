symbol foo() : Bool
symbol goo() : Bool
symbol bar() : Bool

axiom foo_holds: foo()
axiom goo_holds: goo()
axiom foo_and_goo_implies_bar:
  \implies(foo(), \implies(goo(), bar()))

claim bar()
strategy
       apply(
          foo_and_goo_implies_bar,
          apply(foo_holds, fail) | apply(goo_holds, fail)
       )
