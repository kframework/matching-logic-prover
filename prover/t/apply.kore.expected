<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom ax0 : bar ( .Patterns )
    </declaration> <declaration>
      axiom foo_and_goo_implies_bar : \implies ( foo ( .Patterns ) , \implies ( goo ( .Patterns ) , bar ( .Patterns ) ) )
    </declaration> <declaration>
      axiom foo_holds : foo ( .Patterns )
    </declaration> <declaration>
      axiom goo_holds : goo ( .Patterns )
    </declaration> <declaration>
      symbol bar ( .Sorts ) : Bool
    </declaration> <declaration>
      symbol foo ( .Sorts ) : Bool
    </declaration> <declaration>
      symbol goo ( .Sorts ) : Bool
    </declaration>
  </declarations>
</prover>
